libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgress.kt

```diff
@@ -153,7 +160,7 @@ class RouteProgress private constructor(
         private var routeGeometryWithBuffer: Geometry? = null
         private var bannerInstructions: BannerInstructions? = null
         private var voiceInstructions: VoiceInstructions? = null
-        private var currentState: RouteProgressState = RouteProgressState.ROUTE_INVALID
+        private var currentState: RouteProgressState = RouteProgressState.INITIALIZED
```

Comment:
Is it actually `INITIALIZED` at this point? 🤔

Reply:
No, but a nullable `RouteProgressState` wouldn't be ideal either.

I think we should remove the builder pattern and move `RouteProgress` to an internal factory as well - https://github.com/mapbox/mapbox-navigation-android/issues/4414.